server/opts.go

```diff
@@ -193,11 +193,31 @@ func parseCluster(cm map[string]interface{}, opts *Options) error {
 			if opts.ClusterTLSConfig, err = GenTLSConfig(tc); err != nil {
 				return err
 			}
-			// For clusters, we will force strict verification. We also act
+
+			// For clusters, we will force strict verification unless the user
+			// specifically disables this by setting `verify: false`. We also act
 			// as both client and server, so will mirror the rootCA to the
 			// clientCA pool.
-			opts.ClusterTLSConfig.ClientAuth = tls.RequireAndVerifyClientCert
-			opts.ClusterTLSConfig.ClientCAs = opts.ClusterTLSConfig.RootCAs
+			//
+			// Because an unspecified verify defaults to false by default we must
+			// first explicitly set the default to true and then fill in the
+			// user-supplied value if it is set in the config.
+			tc.Verify = true
```

Comment:
I originally considered this as well, but didn't want to change the signature of `parseTLS`. However I realize it isn't an exported function and only used within `opts.go` so there's no harm in it.

I did go for a slightly more general approach, instead of having an argument specifically for `verify` I let it pass in a `TLSConfigOpts` struct directly, so you can set defaults for any field.